### ISSUE TITLE
Basic questions about IoU

### ISSUE THREAD

**Issue by youkangGG** [USTCLH/OccMamba#6]
Thank you for your excellent open-source work! I'm a beginner trying to reproduce your results, but I'm stuck on some basic issues. Could you please point me in the right direction?

Specifically, I ran your pre-trained model and obtained the following results, but I did not see any output regarding IoU and mIoU. Does 'SC_non-empty' represent IoU (the 4th column in Table 2 of the OccMamba paper), while 'SSC_barrier/SSC_bicycle/SSC_bus' represent the per-class IoUs (columns 6–21 in Table 2), and 'SSC_fine_mean' represent mIoU (the 5th column in Table 2)? Is IoU only an evaluation of voxel occupancy and not related to semantic information?

<blockquote>
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6020/6019, 0.7 task/s, elapsed: 8135s, ETA:     0s{'SC_non-empty': 0.345, 'SSC_free': 0.933, 'SSC_barrier': 0.084, 'SSC_bicycle': 0.107, 'SSC_bus': 0.137, 'SSC_car': 0.123, 'SSC_construction_vehicle': 0.098, 'SSC_motorcycle': 0.149, 'SSC_pedestrian': 0.11, 'SSC_traffic_cone': 0.111, 'SSC_trailer': 0.085, 'SSC_truck': 0.132, 'SSC_driveable_surface': 0.241, 'SSC_other_flat': 0.199, 'SSC_sidewalk': 0.18, 'SSC_terrain': 0.177, 'SSC_manmade': 0.164, 'SSC_vegetation': 0.204, 'SSC_mean': 0.144, 'SSC_fine_free': 0.979, 'SSC_fine_barrier': 0.217, 'SSC_fine_bicycle': 0.155, 'SSC_fine_bus': 0.239, 'SSC_fine_car': 0.265, 'SSC_fine_construction_vehicle': 0.159, 'SSC_fine_motorcycle': 0.229, 'SSC_fine_pedestrian': 0.224, 'SSC_fine_traffic_cone': 0.198, 'SSC_fine_trailer': 0.191, 'SSC_fine_truck': 0.237, 'SSC_fine_driveable_surface': 0.37, 'SSC_fine_other_flat': 0.254, 'SSC_fine_sidewalk': 0.254, 'SSC_fine_terrain': 0.254, 'SSC_fine_manmade': 0.28, 'SSC_fine_vegetation': 0.299, 'SSC_fine_mean': 0.239}
</blockquote>

I would be truly grateful if you could kindly shed some light on these questions.

**Comment by USTCLH** [comment]
Your understanding is generally correct, but the IoU for each class in Table 2 of the paper actually corresponds to SSC_fine_(classname) rather than SSC_(classname). Additionally, I noticed that the mIoU you reproduced using the provided weights is only 23.9%. Have you checked whether the version of the network weights and OpenOccupancy labels are aligned?

Moreover, the [code](https://github.com/USTCLH/OccMamba/blob/main/projects/occ_plugin/utils/formating.py#L74) actually includes PrettyTable for displaying the results. You might want to utilize it to make the output more readable and well-organized.

**Comment by youkangGG** [comment]
Thanks! I'll double-check how the network weights match up with the OpenOccupancy labels. Your assistance is highly appreciated :)
